crates/astria-sequencer/src/slinky/marketmap/component.rs

```diff
@@ -0,0 +1,53 @@
+use std::sync::Arc;
+
+use anyhow::{
+    Context,
+    Result,
+};
+use astria_core::protocol::genesis::v1alpha1::GenesisAppState;
+use tendermint::abci::request::{
+    BeginBlock,
+    EndBlock,
+};
+use tracing::instrument;
+
+use super::state_ext::StateWriteExt;
+use crate::component::Component;
+
+#[derive(Default)]
+pub(crate) struct MarketMapComponent;
+
+#[async_trait::async_trait]
+impl Component for MarketMapComponent {
+    type AppState = GenesisAppState;
+
+    #[instrument(name = "MarketMapComponent::init_chain", skip(state))]
+    async fn init_chain<S: StateWriteExt>(mut state: S, app_state: &Self::AppState) -> Result<()> {
```

Comment:
Make this `S: StateWrite` and only import that `StateWriteExt as _` at the head of this file.